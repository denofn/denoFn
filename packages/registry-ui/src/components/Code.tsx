import prettier from "prettier/standalone";
import tsParser from "prettier/parser-typescript";
import { highlight, languages } from "prismjs";
import { css } from "otion";
import React from "react";
import Editor from "react-simple-code-editor";
import "prismjs/components/prism-typescript";
import "prismjs/themes/prism.css";

import { Action } from "./Function";

export const Code = React.memo(function Code({ index, dispatch }: { index?: string; dispatch: (a: Action) => void }) {
  const handleValueChange = (code: string) => {
    dispatch({ type: "index", value: code });
  };

  const handleBlur = () => {
    const formattedCode = prettier.format(`${index}`, {
      semi: true,
      printWidth: 100,
      parser: "typescript",
      plugins: [tsParser],
    });
    dispatch({ type: "index", value: formattedCode });
  };

  return (
    <div
      className={css({
        margin: "1.5rem 0",
      })}
    >
      <div className={css({ margin: "1.5rem 0 0.75rem", fontWeight: "bold" })}>index.ts</div>
      <Editor
        value={`${index}`}
        onValueChange={handleValueChange}
        highlight={(code: string) => highlight(code, languages.ts, "ts")}
        onBlur={handleBlur}
        padding={4}
        className={css({
          fontFamily: "'Fira Code', monospace",
        })}
      />
    </div>
  );
});
