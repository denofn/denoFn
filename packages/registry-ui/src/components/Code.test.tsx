import { render, cleanup } from "@testing-library/react";
import React from "react";

import { Code } from "./Code";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Code dispatch={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
