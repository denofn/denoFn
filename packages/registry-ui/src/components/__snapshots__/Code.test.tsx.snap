// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_di48ht"
  >
    <div
      class="_1szb0m3 _1efi8gv"
    >
      index.ts
    </div>
    <div
      class="_1qnppsc"
      style="position: relative; text-align: left; box-sizing: border-box; padding: 0px; overflow: hidden;"
    >
      <textarea
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="npm__react-simple-code-editor__textarea"
        data-gramm="false"
        spellcheck="false"
        style="margin: 0px; border: 0px; background: none; box-sizing: inherit; display: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant-ligatures: inherit; font-weight: inherit; letter-spacing: inherit; line-height: inherit; tab-size: inherit; text-indent: inherit; text-rendering: inherit; text-transform: inherit; white-space: pre-wrap; word-break: keep-all; overflow-wrap: break-word; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; resize: none; overflow: hidden; padding: 4px 4px 4px 4px;"
      >
        undefined
      </textarea>
      <pre
        aria-hidden="true"
        style="margin: 0px; border: 0px; background: none; box-sizing: inherit; display: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant-ligatures: inherit; font-weight: inherit; letter-spacing: inherit; line-height: inherit; tab-size: inherit; text-indent: inherit; text-rendering: inherit; text-transform: inherit; white-space: pre-wrap; word-break: keep-all; overflow-wrap: break-word; position: relative; pointer-events: none; padding: 4px 4px 4px 4px;"
      >
        <span
          class="token keyword"
        >
          undefined
        </span>
        <br />
      </pre>
      <style
        type="text/css"
      >
        
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}

      </style>
    </div>
  </div>
</DocumentFragment>
`;
